feature/beneficiary/src/main/java/org/mifos/mobile/feature/beneficiary/navigation/BeneficiaryNavGraph.kt

```diff
@@ -0,0 +1,135 @@
+            addBeneficiaryManually = {
+                navController.navigate(
+                    BeneficiaryNavigation.BeneficiaryApplication(
+                        BeneficiaryState.CREATE_MANUAL,
+                        null,
+                    ).route
+                )
+             },
+            openQrScanner = openQrReaderScreen,
+            uploadQrCode = openQrImportScreen
+        )
+
+        beneficiaryDetailRoute(
+            navigateBack = { navController.popBackStack() },
+            updateBeneficiary = { beneficiary ->
+                navController.navigate(
+                    BeneficiaryNavigation.BeneficiaryApplication(
+                        BeneficiaryState.UPDATE,
+                        beneficiary
+                    ).route
+                )
+            }
+        )
+
+        beneficiaryApplicationRoute(
+            navigateBack = { navController.popBackStack() }
+        )
+    }
+}
+
+fun NavGraphBuilder.beneficiaryListRoute(
+    navigateBack: () -> Unit,
+    addBeneficiary: () -> Unit,
+    showBeneficiaryDetail: ( Int, List<Beneficiary>) -> Unit
+) {
+    composable(route = BeneficiaryNavigation.BeneficiaryList.route) {
+        BeneficiaryListScreen(
+            navigateBack = navigateBack,
+            addBeneficiaryClicked = addBeneficiary,
+            onBeneficiaryItemClick = { beneficiaryId, beneficiaryList ->
+                showBeneficiaryDetail(beneficiaryId, beneficiaryList)
+            },
+        )
+    }
+}
+
+@RequiresApi(Build.VERSION_CODES.TIRAMISU)
+fun NavGraphBuilder.addBeneficiaryRoute(
+    navigateBack: () -> Unit,
+    addBeneficiaryManually: () -> Unit,
+    openQrScanner: () -> Unit,
+    uploadQrCode: () -> Unit
+) {
+    composable(route = BeneficiaryNavigation.AddBeneficiary.route) {
+        BeneficiaryScreen(
+            topAppbarNavigateback= navigateBack,
+            addiconClicked= addBeneficiaryManually,
+            scaniconClicked= openQrScanner,
+            uploadIconClicked = uploadQrCode
+        )
+    }
+}
+
+fun NavGraphBuilder.beneficiaryDetailRoute(
+    navigateBack: () -> Unit,
+    updateBeneficiary: ( beneficiary : Beneficiary? ) -> Unit,
+    beneficiary : Beneficiary? = null
+) {
+    composable(route = BeneficiaryNavigation.BeneficiaryDetail(beneficiary).route) {
+        BeneficiaryDetailScreen(
+            navigateBack = navigateBack,
+            updateBeneficiary = updateBeneficiary,
+            beneficiary = beneficiary
+        )
+    }
+}
+
+fun NavGraphBuilder.beneficiaryApplicationRoute(
+    navigateBack: () -> Unit,
+    beneficiary : Beneficiary? = null,
+    beneficiaryState: BeneficiaryState? = null
+) {
+    composable(route = BeneficiaryNavigation.BeneficiaryApplication(beneficiaryState, beneficiary).route) {
+        BeneficiaryApplicationScreen(
+            navigateBack = navigateBack,
+            beneficiary = beneficiary,
+            beneficiaryState = beneficiaryState
+        )
+    }
+}
```

Comment:
This is not the proper way of sending arguments in the compose navigation. 
Use navigation args.